central/graphql/resolvers/nodes.go

```diff
@@ -523,20 +542,26 @@ func (resolver *nodeResolver) PlottedNodeVulnerabilities(ctx context.Context, ar
 }
 
 func (resolver *nodeResolver) Scan(ctx context.Context) (*nodeScanResolver, error) {
-	res, err := resolver.root.wrapNodeScan(resolver.data.GetScan(), true, nil)
+	res, err := resolver.root.wrapNodeScanWithContext(resolver.ctx, resolver.data.GetScan(), true, nil)
+	if resolver.ctx == nil {
+		resolver.ctx = ctx
+	}
 	if err != nil || res == nil {
 		return nil, err
 	}
-	res.ctx = resolver.withNodeScopeContext(ctx)
```

Comment:
Should leave this because node scope would be needed inside nodeScanResolver's sub resolvers. You can use resolver.ctx instead of ctx while adding node scope.